typescript/looker/rtl/nodeTransport.ts

```diff
@@ -172,23 +174,26 @@ export class NodeTransport implements ITransport {
 }
 
 async function parseResponse(contentType: string, res: Response) {
-  if (contentType.match(/application\/json/g)) {
-    try {
-      // return await res.json()
-      const result = await res.body
-      return result instanceof Object ? result : JSON.parse(result)
-    } catch (error) {
-      console.log(res.body)
-      return Promise.reject(error)
+  const mode = responseMode(contentType)
+  let result = await res.body
+  if (mode === ResponseMode.string) {
+    if (!isUtf8(contentType)) {
+      // always convert to UTF-8 from whatever it was
+      result = Buffer.from(result.toString(), 'utf8')
+    }
+    if (contentType.match(/^application\/.*\bjson\b/g)) {
+      try {
+        result = result instanceof Object ? result : JSON.parse(result)
```

Comment:
I think I'm missing something - I thought `result` was always a `Buffer` now (except for maybe on L160 you allow overriding `encoding`... why?) If so, don't we want to `toString()` it and then `JSON.parse()` it before returning it? I'm coming from a strong python/requests bias as I'm working on this same code there right now so I might be totally off on this.

Reply:
I think we covered in person, but basically we can't be **sure** the response isn't already converted to a deserialized object because of the request parameters the user could pass in.